Account/FxAPushMessageHandler.swift

```diff
@@ -61,58 +62,63 @@ extension FxAPushMessageHandler {
 
         // return handle(plaintext: string)
         let deferred = PushMessageResult()
-        RustFirefoxAccounts.startup(prefs: profile.prefs) { fxa in
-            RustFirefoxAccounts.reconfig {
-                fxa.accountManager.deviceConstellation()?.processRawIncomingAccountEvent(pushPayload: string) {
+        RustFirefoxAccounts.startup(prefs: profile.prefs).uponQueue(.main) { _ in
+            RustFirefoxAccounts.reconfig { accountManager in
+                accountManager.deviceConstellation()?.processRawIncomingAccountEvent(pushPayload: string) {
                     result in
-                    guard case .success(let events) = result else {
+                    guard case .success(let events) = result, let firstEvent = events.first else {
+                        let err: PushMessageError
                         if case .failure(let error) = result {
-                            let err = PushMessageError.messageIncomplete(error.localizedDescription)
-                            deferred.fill(Maybe(failure: err))
+                            err = PushMessageError.messageIncomplete(error.localizedDescription)
+                        } else {
+                            err = PushMessageError.messageIncomplete("empty message")
                         }
+                        deferred.fill(Maybe(failure: err))
                         return
                     }
-                    events.forEach { event in
-                        switch event {
-                        case .incomingDeviceCommand(let deviceCommand):
-                            switch deviceCommand {
-                                case .tabReceived(_, let tabData):
-                                    let title = tabData.last?.title ?? ""
-                                    let url = tabData.last?.url ?? ""
-                                    let message = PushMessage.commandReceived(tab: ["title": title, "url": url])
-                                    deferred.fill(Maybe(success: message))
-                            }
-                        case .deviceConnected(let deviceName):
-                            let message = PushMessage.deviceConnected(deviceName)
+                    if events.count > 1 {
+                        // Log to the console for debugging release builds
+                        os_log("%{public}@", log: OSLog(subsystem: "org.mozilla.firefox", category: "firefoxnotificationservice"), type: OSLogType.debug, "Multiple events arrived, only handling the first event.")
+                    }
```

Comment:
While updating the code, I noticed that we only handle the firstEvent. This is how the code always was, but I find that `forEach` loop very misleading, as it implies that multiple events are handled.